src/backends/fixerio/index.js

```diff
@@ -122,15 +122,15 @@ class FixerIoBackend {
 
     let limit
     if (sourceInfo.maxBalance !== undefined) {
-      let balanceIn = parseInt(yield this.getBalance(params.source_ledger))
-      let maxAmountIn = sourceInfo.maxBalance - balanceIn
-      limit = [ maxAmountIn, maxAmountIn * rate ]
+      let balanceIn = yield this.getBalance(params.source_ledger)
+      let maxAmountIn = new BigNumber(sourceInfo.maxBalance).minus(balanceIn)
+      limit = [ maxAmountIn.toNumber(), maxAmountIn.times(rate).toNumber() ]
```

Comment:
Are you certain it's safe to use JS sized numbers here? These values could be up to 64 bits.